framework/tlvf/src/src/BaseClass.cpp

```diff
@@ -35,7 +35,7 @@ bool BaseClass::isInitialized() { return m_init_succeeded; }
  */
 bool BaseClass::buffPtrIncrementSafe(size_t length)
 {
-    if (getBuffRemainingBytes() <= length) {
+    if (getBuffRemainingBytes() < length) {
```

Comment:
Please add a test to tlvf_tests that proves this is needed.

Reply:
I updated the commit message, no need for a test it is definitely a bug - if you have n bytes left and you want to increment by n, then it would fail since n <= n is true.